sycl/plugins/unified_runtime/ur/adapters/cuda/image.cpp

```diff
@@ -0,0 +1,1029 @@
+  // small data types need to be minimised to 4.
+  if (elementSizeBytes < 4) {
+    elementSizeBytes = 4;
+  }
+  UR_ASSERT((elementSizeBytes == 4 || elementSizeBytes == 8 ||
+             elementSizeBytes == 16),
+            UR_RESULT_ERROR_INVALID_VALUE);
+  ur_result_t Result = UR_RESULT_SUCCESS;
+  try {
+    ScopedContext Active(hDevice->getContext());
+    Result =
+        UR_CHECK_ERROR(cuMemAllocPitch((CUdeviceptr *)ppMem, pResultPitch,
+                                       widthInBytes, height, elementSizeBytes));
+  } catch (ur_result_t error) {
+    Result = error;
+  } catch (...) {
+    return UR_RESULT_ERROR_UNKNOWN;
+  }
+
+  return Result;
+}
+
+UR_APIEXPORT ur_result_t UR_APICALL
+urBindlessImagesUnsampledImageHandleDestroyExp(ur_context_handle_t hContext,
+                                               ur_device_handle_t hDevice,
+                                               ur_exp_image_handle_t hImage) {
+  UR_ASSERT(hContext, UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+  UR_ASSERT(hDevice, UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+  UR_ASSERT((hContext->getDevice()->get() == hDevice->get()),
+            UR_RESULT_ERROR_INVALID_CONTEXT);
+  UR_ASSERT(hImage, UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+
+  ur_result_t RetErr = UR_RESULT_SUCCESS;
+  RetErr = UR_CHECK_ERROR(cuSurfObjectDestroy((CUsurfObject)hImage));
+  return RetErr;
+}
+
+UR_APIEXPORT ur_result_t UR_APICALL
+urBindlessImagesSampledImageHandleDestroyExp(ur_context_handle_t hContext,
+                                             ur_device_handle_t hDevice,
+                                             ur_exp_image_handle_t hImage) {
+  UR_ASSERT(hContext, UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+  UR_ASSERT(hDevice, UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+  UR_ASSERT((hContext->getDevice()->get() == hDevice->get()),
+            UR_RESULT_ERROR_INVALID_CONTEXT);
+  UR_ASSERT(hImage, UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+
+  ur_result_t RetErr = UR_RESULT_SUCCESS;
+  RetErr = UR_CHECK_ERROR(cuTexObjectDestroy((CUtexObject)hImage));
+  return RetErr;
+}
+
+UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesImageAllocateExp(
+    ur_context_handle_t hContext, ur_device_handle_t hDevice,
+    const ur_image_format_t *pImageFormat, const ur_image_desc_t *pImageDesc,
+    ur_exp_image_mem_handle_t *phImageMem) {
+  UR_ASSERT(hContext, UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+  UR_ASSERT(hDevice, UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+  UR_ASSERT((hContext->getDevice()->get() == hDevice->get()),
+            UR_RESULT_ERROR_INVALID_CONTEXT);
+  UR_ASSERT(pImageFormat, UR_RESULT_ERROR_INVALID_NULL_POINTER);
+  UR_ASSERT(pImageDesc, UR_RESULT_ERROR_INVALID_NULL_POINTER);
+  UR_ASSERT(phImageMem, UR_RESULT_ERROR_INVALID_NULL_POINTER);
+
+  ur_result_t RetErr = UR_RESULT_SUCCESS;
+
+  // Populate descriptor
+  CUDA_ARRAY3D_DESCRIPTOR array_desc = {};
+
+  RetErr = UR_CHECK_ERROR(urCalculateNumChannels(pImageFormat->channelOrder,
```

Comment:
Same as above, re tracking return value in `RetErr`

Reply:
Changed.